com.microsoft.mrtk.uxcore/Dialog/Dialog.cs

```diff
@@ -2,259 +2,257 @@
-        /// </summary>
-        [Obsolete("Dialog's ConstantViewSize/variable follow distance is deprecated until Dialog is refactored to render correctly when scaled.")]
-        public float FollowMaxDistanceFar
-        {
-            get => followMaxDistanceFar;
-            set => followMaxDistanceFar = value;
-        }
+        private UnityEvent<Dialog> onDismissed = new UnityEvent<Dialog>();
 
-        [SerializeField]
-        [HideInInspector]
-        private float followDefaultDistanceFar = 1.2f;
+        /// <inheritdoc />
+        public UnityEvent<Dialog> OnDismissed => onDismissed;
 
-        /// <summary>
-        /// The default distance setting on the follow solver attached to this dialog in far interaction placement mode.
-        /// </summary>
-        [Obsolete("Dialog's ConstantViewSize/variable follow distance is deprecated until Dialog is refactored to render correctly when scaled.")]
-        public float FollowDefaultDistanceFar
-        {
-            get => followDefaultDistanceFar;
-            set => followDefaultDistanceFar = value;
-        }
+        #endregion
 
-        [SerializeField]
-        [HideInInspector]
-        private bool placeForNearInteraction = true;
+        #region Private helper fields
 
-        /// <summary>
-        /// Use placement optimized for near interaction instead of far interaction.
-        /// </summary>
-        public bool PlaceForNearInteraction
+        private bool hasDismissed = false;
+
+        #endregion
+
+        /// <inheritdoc />
+        public IDialog SetHeader(string header)
         {
-            get => placeForNearInteraction;
-            set
-            {
-                if (placeForNearInteraction != value)
-                {
-                    SetInteractionMode(value);
-                    placeForNearInteraction = value;
-                }
-            }
+            this.header = header;
+            return this;
         }
 
-        /// <summary>
-        /// Generates buttons - Must parent them under buttonParent!
-        /// </summary>
-        protected abstract void GenerateButtons();
-
-        /// <summary>
-        /// This is called after the buttons are generated and
-        /// the title and message have been set.
-        /// Perform here any operations that you'd like
-        /// Lays out the buttons on the dialog
-        /// E.g. using an ObjectCollection
-        /// </summary>
-        protected abstract void FinalizeLayout();
+        /// <inheritdoc />
+        public IDialog SetBody(string body)
+        {
+            this.body = body;
+            return this;
+        }
+        
+        /// <inheritdoc />
+        public IDialog SetPositive(string label, UnityAction<DialogButtonEventArgs> action)
+        {
+            if (label == null) { return this; }
+            positiveButton.Label.text = label;
+            positiveAction = new DialogButtonEvent();
```

Comment:
Ope, nevermind, I'm using the nullishness of the event to determine whether there's any listeners or not, because Unity lacks an API for checking the number of non-persistent listeners.